backend/core/rule/tree.py

```diff
@@ -0,0 +1,72 @@
+import yaml
+from core.rule.functions import *
+
+
+class TreeLoader(object):
+    def __init__(self, rule_name, sugang_list):
+        f = open(rule_name, 'r', encoding='utf-8')
+        self.tree = yaml.load(f)
+        f.close()
+        self.sugang_list = sugang_list
+        self.base_node = TreeNode(None, and_func())
+        self.load_tree(self.tree, self.base_node)
+
+    def load_tree(self, current_tree, previous_node): # current_tree는 리스트!
+        for course in current_tree:
+            if type(course) is str:
+                previous_node.add_children(TreeNode(course, None, self.sugang_list))
+            elif type(course) is dict:
+                args = course.get('args', [])
+                current_node = TreeNode(None, make_func(course['func'])(*args))
+                self.load_tree(course['courses'], current_node)
+                previous_node.add_children(current_node)
+
+    def eval_tree(self):
+        self.base_node.eval_children()
+        return self.base_node.data
+
+
+class TreeNode(object):
+    def __init__(self, data, func, *args):
+        self.data = data
+        self.func = func
+        self.is_course = False
+        if type(data) is str:
+            self.is_course = True
+        if self.is_course is False:
```

Comment:
`self.is_course` itself is boolean type. I think changing order of `if` and `else` clause would be better.